src/Game/UI/Gumps/WorldMapGump.cs

```diff
@@ -476,29 +503,43 @@ public override bool Draw(UltimaBatcher2D batcher, int x, int y)
 
         private void DrawAll(UltimaBatcher2D batcher, int gX, int gY, int halfWidth, int halfHeight)
         {
-            foreach (Mobile mob in World.Mobiles)
+            if (_showCoordinates)
             {
-                if (mob == World.Player)
-                    continue;
+                _sb.Clear();
+                _sb.AppendLine($"{World.Player.X}, {World.Player.Y} ({World.Player.Z})");
```

Comment:
using SB is a bit overhead here.
Just store player.x,y,z into 3 vars and check if they change. When they change update the string

Reply:
Updated PR based on your feedback. Thanks!